src/condor/htcondor_helper.py

```diff
@@ -2,174 +2,332 @@
 
-                # get job statuses
-                ads = self.schedd.query(
-                    constraint=f"ClusterId == {cluster_id}",
-                    projection=["ProcID", "JobStatus"]
-                )
-                
-                if not ads:
-                    logger.warning(f"No jobs found for cluster {cluster_id}")
-                    return False
-                
-                # Check if all jobs completed
-                if all(ad.get("JobStatus") == 4 for ad in ads):
-                    logger.info(f"All jobs in cluster {cluster_id} completed successfully")
-                    return True
-                
-                # Log status counts
-                status_counts = {}
-                for ad in ads:
-                    status = ad.get("JobStatus", 0)
-                    status_counts[status] = status_counts.get(status, 0) + 1
+                        self.logger.debug(f"Querying status for cluster {cluster_id}")
+                        ads = self.schedd.query(
+                            constraint=f"ClusterId == {cluster_id}",
+                            projection=["ProcID", "JobStatus", "HoldReason", "HoldReasonCode"]
+                        )
                 
-                status_desc = {
-                    1: "Idle", 
-                    2: "Running", 
-                    3: "Removed",
-                    4: "Completed", 
-                    5: "Held", 
-                    6: "Transferring Output",
-                    7: "Suspended"
-                }
-                
-                status_msg = ", ".join(f"{status_desc.get(k, f'Unknown({k})')}: {v}" 
-                                        for k, v in status_counts.items())
-                logger.info(f"Cluster {cluster_id} status: {status_msg}. Total wait time: {total_wait} seconds")
-                
-                # for some reason, jobs get 'held' when submitted instead of 'idle'. not sure why.
-                # they do seem to go to 'idle' shortly thereafter though.
-                # 'held' jobs basically also mean 'failed' jobs.
-                # so we will say that if a job is not 'idle' or 'running' after 2 minutes, it has failed.
-                # it's a bit hacky...
-                if total_wait >= 120:
-                    job_is_idle = status_counts.get(1, 0) > 0
-                    job_is_running = status_counts.get(2, 0) > 0
-                    if not job_is_idle and not job_is_running:
-                        raise Exception(f"Cluster {cluster_id} timed out (job was held too long)")
-
-                time.sleep(check_interval)
-                total_wait += check_interval
+                    if not ads:
+                        self.logger.warning(f"No jobs found for cluster {cluster_id}")
+                        return False
+                    
+                    # Check if all jobs completed
+                    completed_jobs = [ad for ad in ads if ad.get("JobStatus") == 4]
```

Comment:
consider simplifying with 'if all' (please review the code that this PR removes)